tests/testthat/test-integration_metrics.R

```diff
@@ -0,0 +1,116 @@
+# generate testing data
+set.seed(1665)
+merged_sce <- sim_sce(n_cells = 303, n_genes = 100, n_empty = 0)
+batches <- rep(c("a", "b", "c"), each = 101)
+barcodes <- rownames(colData(merged_sce))
+
+# set up colData rownames as batch-barcode
+new_rownames <- glue::glue("{batches}-{barcodes}")
+rownames(colData(merged_sce)) <- new_rownames
+
+# Add "sample" to colData
+colData(merged_sce)$sample <- batches
+
+# add PCs for testing (exact numbers don't matter)
+# make a 300x100 matrix
+reducedDim(merged_sce, "PCA") <- matrix(runif(303 * 100, min = 0, max = 100), nrow = 303)
+
+# grab pcs and batches
+pcs <- reducedDim(merged_sce, "PCA")
+batches <- merged_sce$sample
+
+test_that("`filter_pcs` works as expected", {
+
+  # expect rownames of returned pcs to be the same as the batch column
+  named_pcs <- filter_pcs(pcs, batches)
+  expect_true(all.equal(rownames(named_pcs), batches))
+
+})
+
+test_that("`filter_pcs` removes NAs as expected", {
+
+  batches[1:5] <- NA # set a subset of batch labels to NAs
+
+  # expect rownames to be equal to only the batches without NA
+  named_pcs <- filter_pcs(pcs, batches)
+  expect_true(all.equal(rownames(named_pcs), batches[!is.na(batches)]))
+  expect_false(length(rownames(named_pcs)) == length(batches))
+})
+
+test_that("`filter_pcs` fails without batch labels", {
+
+  all_na_batches <- NA
+  expect_error(scpcaTools:::filter_pcs(pcs, all_na_batches))
```

Comment:
I don't think you need to name `scpcaTools` here?
```suggestion
  expect_error(filter_pcs(pcs, all_na_batches))
```

Reply:
No I don't, I must have forgotten to remove it from when I was testing things locally.